modules/mutect2/2.0/mutect2.smk

```diff
@@ -123,18 +123,34 @@ rule _mutect2_get_sm:
 # The first function loads the wildcard-containing file path and additional args from the config. 
 # The second replaces wildcards with those used in the rule. 
 def _mutect2_get_interval_cli_arg(
-    vcf_in = config["lcr-modules"]["mutect2"]["inputs"]["candidate_positions"], 
```

Comment:
I can't comment on code that's unchanged, but the Mutect2 has long been awaiting a permanent bug fix for the instance where the input candidate_positions file doesn't cover all of the chromosomes in `reference_files("genomes/{genome_build}/genome_fasta/main_chromosomes.txt")`. I think we should also cover the possibility that the cap_space bed file doesn't span all chromosomes (I have a capture panel for which this is true).  I have a few thoughts on how this could be handled but either way I think somehow it needs to precede the checkpoint.

Reply:
I can work on these changes within this PR, and implement this bug fix. I haven't run mutect2 explicitly but can read the documentation on how to deal with this and come up with some suggestions. Will work on it

Reply:
Is this the bug where MuTect2 errors out because basically there are no input regions specified (i.e. the overlap between all the files/regions is empty)

Reply:
Yes, Chris - it is about this bug

Reply:
I think me and Haya came up with a solution, but I'm not sure what happened to that

Reply:
The checkpoint here has been modified to intersect 3 files of  a) main chromosomes, b) candidate positions, and c) capture space (which is set to main chromosomes in case of non-capture samples). The intersect of chrs in these 3 files will dictate mutect2 on which chromosomes to run ✅